tsl/src/continuous_aggs/refresh.c

```diff
@@ -374,3 +374,54 @@ continuous_agg_refresh_internal(const ContinuousAgg *cagg,
 			 "continuous aggregate \"%s\" is already up-to-date",
 			 NameStr(cagg->data.user_view_name));
 }
+
+/*
+ * Refresh all continuous aggregates on a hypertable.
+ *
+ * The refreshing happens in a single transaction. For this to work correctly,
+ * there must be no new invalidations written in the refreshed region during
+ * the refresh. Therefore, the caller is responsible for proper locking to
+ * ensure there are no invalidations.
```

Comment:
There is some locking of cagg tables in the function below. What kind of locks should the user set and on what? Could you add that to the comment?